tools/fifo.c

```diff
@@ -22,6 +22,29 @@ uint16_t fifo_advance(uint16_t pos)
 	return (pos >= FIFO_MAX) ? 0 : pos;
 }
 
+uint16_t fifo_seek(int16_t pos, int16_t steps)
+{
+	pos += steps;
+	if (pos >= FIFO_MAX) {
```

Comment:
if steps > 2 * FIFO_MAX, it will still be out of range, probably replace with a while loop?